### PR TITLE
Increase timeout, remove double close.

### LINKED ISSUE #886: TestAppClientDetails is flaky
```
2016/01/29 21:36:15 Publish loop for http://127.0.0.1:47115 starting
2016/01/29 21:36:15 Publish loop for http://127.0.0.1:47115 exiting
--- FAIL: TestAppClientDetails (0.10s)
	app_client_internal_test.go:147: timeout
FAIL
FAIL	github.com/weaveworks/scope/probe/appclient	0.134s
```

### PR DESCRIPTION
**Pull request by tomwilkie**
Fix #886

### CODE REVIEW

File: probe/appclient/app_client_internal_test.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func TestAppClientDetails(t *testing.T) {
 
 	select {
 	case <-done:
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(200 * time.Millisecond):
```
Comment (paulbellamy) [inline probe/appclient/app_client_internal_test.go]: crank it up. Why not a full second?

Comment (paulbellamy) [inline probe/appclient/app_client_internal_test.go]: Also, why not the one above in `TestAppClientPublishInternal`?

**Comment by paulbellamy** [comment]
A better solution would seem to be to make `AppClient.Publish` block until the publish has finished/failed, then you don't need polling/timeouts in the tests.

**Comment by tomwilkie** [comment]
> AppClient.Publish block

That would be problematic from the POV of the code - publish is supposed to be best effort, and we don't want it to block the caller.  For instance, when publishing to two apps, do you wait for both to ACK?  If so you have the ability for a slow app to starve a faster one, for instance.  In essence it would make MultiClient's job harder.

Anyway I misdiagnosed this, it was the first few reports getting dropped.  I've increased the number and added a small sleep to give go time to spin up the right goroutines.

**Comment by paulbellamy** [comment]
Then do `go AppClient.Publish()` where you don't care about the response. Not that much harder.

**Pull request merged**